UOP1_Project/Assets/Scripts/StateMachineTest/ScriptableHungerAction.cs

```diff
@@ -0,0 +1,19 @@
+﻿using DeivSky.StateMachine;
+using DeivSky.StateMachine.Scriptables;
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "EnableHunger", menuName = "State Machines/Tests/Actions/Enable Hunger")]
+public class ScriptableHungerAction : ScriptableStateAction<HungerAction> { }
+
+public class HungerAction : StateAction
+{
+	private HungerComponent _hungerComponent;
+
+	public override void Awake(StateMachine stateMachine)
+		=> _hungerComponent = stateMachine.GetComponent<HungerComponent>();
+
+	public override void Perform() { }
+
+	public override void OnStateEnter() => _hungerComponent.ToggleHunger(true);
+	public override void OnStateExit() => _hungerComponent.ToggleHunger(false);
```

Comment:
Why did you make all functions into lambdas?

Reply:
Not lambdas, expression body definitions. 
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/lambda-operator

It's exactly the same as writing
``` csharp
public override void OnStateEnter()
{
    _hungerComponent.ToggleHunger(true);
}
```
Just less boilerplate.

Reply:
Yeah, I don't know, I think it's quite confusing to have half of the codebase with methods expressed with curly braces, and half of it using expression body definitions.
Personally I would avoid them, I generally like the "framing" that curly braces give to a function's body (even if it's a one liner), and they are also kind of obscure for beginner users.

Reply:
Yeah it's personal preference. I did ask about it in the coding standards thread a while ago, no one said much about it though. I don't mind switching to brackets, should we add that to the document?

Reply:
Expression bodied members have been around for half a decade (C# 6 - 2015) hardly obscure. One of the most maddening things about Unity examples is how they seem to refuse to use new, simple syntax idioms that reduce visual noise and make things easier to read. Pattern matching is the next hill that Unity evangelists are going to have to climb and need to push into common usage.

Part of this used to come from Unity being so far behind (pre 3.5) Mono releases, so old syntax was commonplace (like "get {}; set{}; instead of "{get;set}"  for simple Props) but now that Unity is going to track C# at a quicker pace, Unity "best practices" should keep up with modern usage if possible.

Using R# or Rider it is simple to switch everything to expressions bodied methods for consistency.

Reply:
> simple syntax idioms that reduce visual noise and make things easier to read

I don't know if I find expression-bodied members easier to read, less characters don't necessarily mean more readable. In fact I find them less readable!

But as @DeivSky said, it's a matter of preference so I don't mind too much if he wants to keep them in the state machine, but I personally wouldn't love to see them all over the codebase.